examples/simple/igraph_assortativity.c

```diff
@@ -0,0 +1,35 @@
+#include <igraph.h>
+#include <stdio.h>
+
+int main(){
+    igraph_t g;
+    igraph_integer_t nodes = 100000, A = 0, power = 1, m = 1;
+    igraph_vector_t degree;
+    igraph_vector_init(&degree, 0);
+    
+    igraph_rng_seed(igraph_rng_default(), 42);
+    printf("Barabasi-Albert network with 100000 nodes\n\n");
+
+    for (int i = 0; i < 5; i++) {
+        igraph_real_t assortativity;
+
+        /* Generate undirected graph with 100000 nodes */
+        igraph_barabasi_game(&g, nodes, power, m, NULL, IGRAPH_UNDIRECTED, A, IGRAPH_UNDIRECTED, IGRAPH_BARABASI_PSUMTREE, /* start from */ NULL);
+
+        /* Get degree sequence */
+        igraph_strength(&g, &degree, igraph_vss_all(), IGRAPH_ALL, /* consider self-loops */ IGRAPH_LOOPS, /* no weights */ NULL);
+       
+        igraph_assortativity(&g, &degree, NULL, &assortativity, /* ignore edge directions */ IGRAPH_UNDIRECTED);
```

Comment:
Let us change this to use `igraph_assortativity_degree()` and make this into an example for that function instead. Then we can get rid of `igraph_strength()` completely.

I am concerned that if people see this example, in its current form, they will believe that this is the usual way to compute degree assortativity (the most common use case). In fact, they should be using `igraph_assortativity_degree()` for that task, as it's more convenient.